### DISCUSSION TITLE
Season Models & Current Data

### DISCUSSION THREAD

**Discussion by jfacowns** [kyleskom/NBA-Machine-Learning-Sports-Betting#56] [General]
First off, this is a really cool project and I’m really enjoying going through it. 

I do have a few questions however.

The repo includes data all the way back to the 2007-08 season but in the notes, you make reference to “the highest levels of validation accuracy when the training dataset started from the 2012 − 2013 season rather than from the 2007 − 2008 season.” 

Can I therefore assume that the model is using only 2012-13 until last year (2021-22) ? 

Also wondering what the reason might be for not including games that have already happened this season (2022-23)? Would this not be the most predictive data to use in a model given that players and coaches change teams every year? 

I’ve used the Get_data.py script to try and pull this years data after changing the year variables to 2022-23. The script runs well as expected and appears to be pulling data but once it’s finished, I see no output. Looks like it’s supposed to create a new season folder in Team-Data and then populate.

Again, this is really cool and I’m looking forward to seeing how this progresses. Shoutout to those who found a way to integrate different sportsbooks and pull odds. Awesome stuff.

**Comment by kyleskom** [comment]
1.) That note was added after I trained the models some time back, I added that from a paper I read to revise as some point in the future. The training data used goes back to 07.

2.) The way I originally implemented things its not super easy to automatically get must up to date date and retrain. Would like to do that at some point just wasn't a priority when first starting thing.

3.) get data should just save into the team data folder. It's not super clean and this is where a lot of the messiness / bad practices are.

**Reply by dm-itriy** [reply]
I also spent some time digging and appreciate the work that you have done. However I do think that the biggest flaw is the fact that you don't use intra-season games which would have substantial changes in your models if you want to have a higher accuracy to make money with. In addition, the next biggest flaw is not accounting for injuries on teams which is something that I would think would be hard to change of course and is something that the user would need to be conscious about if they wanted to use your models to place bets, would highly recommend the first change and the second just tell the users in the read me that this does not consider injuries on teams, this would be a great next step after the first.

**Reply by kyleskom** [reply]
Yes that is correct. All things I have considered but don't quite have the time for. Im always open for prs. Still h ope to have this at some point in the future.
